mteb/abstasks/AbsTaskBitextMining.py

```diff
@@ -100,3 +115,29 @@ def _evaluate_subset(
 
     def _add_main_score(self, scores) -> None:
         scores["main_score"] = scores[self.metadata.main_score]
+
+    def _calculate_metrics_from_split(
+        self, split: str, lang: str | None = None
+    ) -> BitextDescriptiveStatistics:
+        pairs_cols = self.get_pairs(self.parallel_subsets)
+        if lang:
+            if self.parallel_subsets:
+                sent_1, sent_2 = lang.split("-")
+                sentence1 = self.dataset[split][sent_1]
+                sentence2 = self.dataset[split][sent_2]
+            else:
+                sent_1, sent_2 = pairs_cols[0]
+                sentence1 = self.dataset[lang][split][sent_1]
+                sentence2 = self.dataset[lang][split][sent_2]
+        else:
+            sent_1, sent_2 = pairs_cols[0]
+            sentence1 = self.dataset[split][sent_1]
+            sentence2 = self.dataset[split][sent_2]
+        total_s1_len = sum([len(s1) for s1 in sentence1])
+        total_s2_len = sum([len(s2) for s2 in sentence2])
+
+        return {
+            "average_sentence1_length": total_s1_len / len(sentence1),
+            "average_sentence2_length": total_s2_len / len(sentence2),
+            "num_samples": len(sentence1),
+        }
```

Comment:
Should return BitextDescriptiveStatistics(...)

Reply:
seems to be a general correction

Reply:
TypedDict does not support init like this. I can make like
```
TypedDict('BitextDescriptiveStatistics', ...)
```
or
```
result: BitextDescriptiveStatistics = ...
```
Maybe we should use dataclass instead of TypedDict?

Reply:
A data class works just fine as well. 

Though TypedDict does support init:

```
from typing import TypedDict

class Movie(TypedDict):
    name: str

t = Movie(name='The Matrix') # runs without issue
```

Reply:
Hm. Maybe I missed that when reading documentation